src/commands/registerCommands.ts

```diff
@@ -108,4 +109,75 @@ async function registerValidationCommands(context: ExtensionContext): Promise<vo
         window.showErrorMessage('Error during XML validation: ' + error.message);
       });
   }));
+}
+
+/**
+ * Register commands used for associating grammar file (XSD,DTD) to a given XML file
+ *
+ * @param context the extension context
+ */
+function registerAssociateGrammarCommands(context: ExtensionContext, languageClient: LanguageClient) {
+  context.subscriptions.push(commands.registerCommand(CommandConstants.SELECT_FILE, async (uriString: string, bindingType: string) => {
+    // A click on Bind with... has been processed in the XML document which is not bound to a grammar
+    const documentURI = Uri.parse(uriString);
+    // Open a dialog to select the XSD, DTD to bind.
+    const options = createDialogOptions(bindingType);
+    const fileUri = await window.showOpenDialog(options);
+    if (fileUri && fileUri[0]) {
+      // The XSD, DTD has been selected, get the proper syntax for binding this grammar file in the XML document.
+      const identifier = TextDocumentIdentifier.create(documentURI.toString());
+      const grammarURI = fileUri[0];
+      try {
+        const result = await commands.executeCommand(CommandConstants.EXECUTE_WORKSPACE_COMMAND, CommandConstants.ASSOCIATE_GRAMMAR_INSERT, identifier, grammarURI.toString(), bindingType);
```

Comment:
You don't need to call `EXECUTE_WORKSPACE_COMMAND`, you should be able to call `ASSOCIATE_GRAMMAR_INSERT` directly

Reply:
Are you sure? `EXECUTE_WORKSPACE_COMMAND `means that it's a XML command (from the server). I did a refactoring by splittung CommandConstants to CLientCommandConstants and ServerCommandConstants to avoid this confusion.

Reply:
I deleted the first parameter (`ClientCommandConstants.EXECUTE_WORKSPACE_COMMAND`) in the `commands.executeCommand` function call, and the bind to file functionality still works.